### PR TITLE
feat(cloud): add task rpc client with interpreter

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
1. add rpc protobuf definition on each api which would be used to connect to cloud control plane.
2. add interpreter logic bring necessary role and tenant information for each kind of task request.
3. support to `execute task` by describing task and use current role to execute task sql.
4. complete mock testing server written in python
5. write sqllogic test around all corner cases in interpreter and verify ites results.
6. add ci for continuous quality assurance

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13452)

### CODE REVIEW

File: src/common/cloud-control/proto/task.proto [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+message DescribeTaskResponse {
+  Task task = 1;
+  optional TaskError error = 2;
+}
+message ShowTasksRequest {// every owner has a roles list like ["role1", "role2"], list tasks in the role list
+  string tenant_id = 1;
+  string name_like = 2;
+  int32 result_limit = 4;
+  repeated string owners = 5;   // 根据ownership role拿到的列表，直接在cloud control里面查
```
Comment (BohuTANG) [inline src/common/cloud-control/proto/task.proto:88]: Please comments with English.

File: src/common/cloud-control/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[package]
+name = "common-cloud-control"
```
Comment (BohuTANG) [inline src/common/cloud-control/Cargo.toml:2]: Why put `common-cloud` to `common`?

Comment (BohuTANG) [reply src/common/cloud-control/Cargo.toml:2]: It's a crate of `query`?

Comment (ZhiHanZ) [reply src/common/cloud-control/Cargo.toml:2]: I think it is not purely depend on databend-query as it is an individual RPC client, thus we could put it under common directory more better modularity

Comment (BohuTANG) [reply src/common/cloud-control/Cargo.toml:2]: That's ok

**Review by everpcpc — APPROVED** [review]

File: src/common/cloud-control/src/cloud_api.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+use common_base::base::GlobalInstance;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::task_client::TaskClient;
+
+pub const CLOUD_REQUEST_TIMEOUT_SEC: u64 = 180; // 180 seconds
+
+// commonly used metadata
+pub const TENANT_ID: &str = "X-DATABEND-TENANT-ID";
```
Comment (everpcpc) [inline src/common/cloud-control/src/cloud_api.rs:27]: `X-DATABEND-TENANT` to keep consistent with other headers?

File: tests/cloud-control-server/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+## Databend Cloud Control grpc Server Tests
+
+
+### How to run
```
Comment (BohuTANG) [inline tests/cloud-control-server/README.md:4]: https://github.com/datafuselabs/databend/pull/13482

Please use the snake name:`cloud_control_server`.

**Review by BohuTANG — APPROVED** [review]
Looks good, Let's merge.
